===================================================================
RCS file: /cvs/src/src/bfd/elflink.c,v
retrieving revision 1.420
retrieving revision 1.420.2.1
diff -u -r1.420 -r1.420.2.1
--- src/bfd/elflink.c	2011/09/16 01:17:16	1.420
+++ src/bfd/elflink.c	2011/09/29 05:40:44	1.420.2.1
@@ -11667,9 +11667,10 @@
 	continue;
 
       /* Keep debug and special sections like .comment when they are
-	 not part of a group.  */
+	 not part of a group, or when we have single-member groups.  */
       for (isec = ibfd->sections; isec != NULL; isec = isec->next)
-	if (elf_next_in_group (isec) == NULL
+	if ((elf_next_in_group (isec) == NULL
+	     || elf_next_in_group (isec) == isec)
 	    && ((isec->flags & SEC_DEBUGGING) != 0
 		|| (isec->flags & (SEC_ALLOC | SEC_LOAD | SEC_RELOC)) == 0))
 	  isec->gc_mark = 1;
